superbench/benchmarks/micro_benchmarks/cuda_nccl_bw_performance.py

```diff
@@ -0,0 +1,202 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+"""Module of the NCCL performance benchmarks."""
+
+import os
+import re
+
+from superbench.common.utils import logger
+from superbench.benchmarks import BenchmarkRegistry, Platform, ReturnCode
+from superbench.benchmarks.micro_benchmarks import MicroBenchmarkWithInvoke
+
+
+class CudaNcclBwBenchmark(MicroBenchmarkWithInvoke):
+    """The NCCL bus bandwidth performance benchmark class."""
+    def __init__(self, name, parameters=''):
+        """Constructor.
+
+        Args:
+            name (str): benchmark name.
+            parameters (str): benchmark parameters.
+        """
+        super().__init__(name, parameters)
+
+        self._bin_name = 'all_reduce_perf'
+        self.__algorithms = {
+            'allreduce': 'all_reduce_perf',
+            'allgather': 'all_gather_perf',
+            'broadcast': 'broadcast_perf',
+            'reduce': 'reduce_perf',
+            'reducescatter': 'reduce_scatter_perf'
+        }
+
+    def add_parser_arguments(self):
+        """Add the specified arguments."""
+        super().add_parser_arguments()
+
+        self._parser.add_argument(
+            '--algo',
+            type=str,
+            nargs='+',
+            default=list(self.__algorithms.keys()),
+            help='Nccl algorithms to benchmark. E.g. {}.'.format(' '.join(list(self.__algorithms.keys()))),
+        )
+        self._parser.add_argument(
+            '--gpu_count',
+            type=int,
+            default=8,
+            help='The count of GPUs used for benchmarking.',
```

Comment:
it's confusing, this parameter means ngpu per thread?

Reply:
yes, changed the comment